src/framework/ui/dropdown/droppdownItem.component.tsx

```diff
@@ -0,0 +1,195 @@
+/**
+ * @license
+ * Copyright Akveo. All Rights Reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ */
+
+import React from 'react';
+import {
+  GestureResponderEvent,
+  StyleSheet,
+  TextStyle,
+  TouchableOpacity,
+  TouchableOpacityProps,
+  ViewProps,
+  View,
+} from 'react-native';
+import {
+  Interaction,
+  styled,
+  StyledComponentProps,
+  StyleType,
+} from '@kitten/theme';
+import {
+  Text,
+  TextProps,
+} from '../text/text.component';
+import { CheckBox } from '../checkbox/checkbox.component';
+import { TouchableTypeReturningProps } from '../support/typings';
+
+type TextElement = React.ReactElement<TextProps>;
+type DefaultItemElement = React.ReactElement<TouchableOpacityProps>;
+type MultiSelectItemElement = React.ReactElement<ViewProps>;
+
+export interface DropdownItemType {
+  text: string;
+  textStyle?: TextStyle;
+  disabled?: boolean;
+  items?: DropdownItemType[];
+}
+
+export interface ComponentProps {
+  item: DropdownItemType;
+  selected?: boolean;
+  multiSelect?: boolean;
+  size?: string;
+}
+
+export type DropdownItemProps = ComponentProps & StyledComponentProps & TouchableTypeReturningProps<DropdownItemType>;
+
+class DropdownItemComponent extends React.Component<DropdownItemProps> {
+
+  static styledComponentName: string = 'DropdownItem';
+
+  private onPress = (event: GestureResponderEvent) => {
+    const { item, onPress } = this.props;
+
+    this.props.dispatch([]);
+    onPress(item, event);
+  };
+
+  private onMultiSelectItemPress = (value: boolean): void => {
+    this.onPress(null);
+  };
+
+  private onPressIn = (event: GestureResponderEvent) => {
+    this.props.dispatch([Interaction.ACTIVE]);
+
+    if (this.props.onPressIn) {
+      this.props.onPressIn(this.props.item, event);
+    }
+  };
+
+  private onPressOut = (event: GestureResponderEvent) => {
+    this.props.dispatch([]);
+
+    if (this.props.onPressOut) {
+      this.props.onPressOut(this.props.item, event);
+    }
+  };
+
+  private onLongPress = (event: GestureResponderEvent) => {
+    if (this.props.onLongPress) {
+      this.props.onLongPress(this.props.item, event);
+    }
+  };
+
+  private getComponentStyle = (source: StyleType): StyleType => {
+    const {
+      textColor,
+      textFontSize,
+      textFontWeight,
+      textLineHeight,
+      textMarginHorizontal,
+      selectedBackgroundColor,
+      selectedTextColor,
+      ...containerStyles
+    } = source;
+
+    return {
+      container: containerStyles,
+      text: {
+        color: textColor,
+        fontSize: textFontSize,
+        fontWeight: textFontWeight,
+        lineHeight: textLineHeight,
+        marginHorizontal: textMarginHorizontal,
+      },
+      containerSelected: {
```

Comment:
Looks like this can be implemented with Eva states

Reply:
Done